heron/statemgrs/src/java/com/twitter/heron/statemgr/localfs/LocalFileSystemStateManager.java

```diff
@@ -31,12 +33,50 @@
 import com.twitter.heron.proto.tmaster.TopologyMaster;
 import com.twitter.heron.spi.common.Config;
 import com.twitter.heron.spi.common.Keys;
+import com.twitter.heron.spi.statemgr.Lock;
 import com.twitter.heron.spi.statemgr.WatchCallback;
 import com.twitter.heron.statemgr.FileSystemStateManager;
 
 public class LocalFileSystemStateManager extends FileSystemStateManager {
   private static final Logger LOG = Logger.getLogger(LocalFileSystemStateManager.class.getName());
 
+  /**
+   * Local filesystem implementation of a lock that mimics the file system behavior of the
+   * distributed lock. Only intended for local mode testing, not production. Not guaranteed to be
+   * truly atomic due to limitations in the underlying file system read/write semantics.
+   */
+  private final class FileSystemLock implements Lock {
+    private String path;
+
+    private FileSystemLock(String path) {
+      this.path = path;
+    }
+
+    @Override
```

Comment:
for local file system, there is a file system call that you can use open system call with OCREAT - that automatically creates if a local file does not exist or fails if the file exists. It might be better solution, I think